internal/authorization/resource_manager.go

```diff
@@ -0,0 +1,80 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+)
+
+var ErrNoUserInContext = fmt.Errorf("user not authenticated or requests context broken, cannot perform operation")
+
+func (a *Authorizer) createTuple(ctx context.Context, resourceID string) error {
+	user, ok := ctx.Value(USER_CTX).(*User)
+	if !ok {
+		return ErrNoUserInContext
+	}
+
+	_, err := a.wpool.Submit(
+		a.client.WriteTuple(context.Background(), "user:"+user.ID, CAN_VIEW, resourceID),
+		nil,
+		nil,
+	)
+	return err
+}
+
+func (a *Authorizer) deleteTuple(ctx context.Context, resourceID string) error {
+	user, ok := ctx.Value(USER_CTX).(*User)
+	if !ok {
+		return ErrNoUserInContext
+	}
+
+	_, err := a.wpool.Submit(
+		a.client.WriteTuple(context.Background(), "user:"+user.ID, CAN_VIEW, resourceID),
+		nil,
+		nil,
+	)
```

Comment:
issue: the first parameter to the `Submit` must be a callable, either a 
`func()` or a `func() any`

I'm pretty sure this doesn't work in parallel but just invokes the write tuple right here.
This is supposed to be like

```suggestion
	_, err := a.wpool.Submit(
	        func() any {
	            return a.client.WriteTuple(context.Background(), "user:"+user.ID, CAN_VIEW, resourceID)
	        },
		nil,
		nil,
	)
```

Reply:
at the same time the function should inspect the result and log if error

Reply:
thanks, fixed